README.md

```diff
@@ -1 +1,178 @@
+
+#### Dependencies
+
+`bemhtml` <br>
+`bemtree` <br>
+`$q` <br>
+`$log`
+
+#### Usage
+
+`ngbem(bemjson, useBemtree);`
+
+##### Arguments
+
+Param      | Type      | Details
+-----------|-----------|--------
+bemjson    | `Object`  | Object in `BEMJSON` format to process.
+useBemtree | `Boolean` | If set to `true` process `BEMJSON` with `BEMTREE` first of all.
+
+##### Returns
+
+`Promise` - Promise that will be resolved with HTML string processes from `bemjson`.
+
+## Directives
+
+### ngBem
+
+Directive renders provided `BEMJSON` using `BEMHTML` and optionally `BEMTREE` templates.
+`BEMJSON` has access to scope since is treated as [Angular Expression](https://docs.angularjs.org/guide/expression).
+Rendered html initialized with [`bemdom`'s `init()`](https://bem.info/libs/bem-core/v2.7.0/desktop/i-bem/jsdoc/#jsdoc-init-1) method and compiled with Angular [$compile](https://docs.angularjs.org/api/ng/service/$compile) service.
+List of scope variables could be provided to observe changes, which will lead to rerendering provided `BEMJSON`.
+
+#### Dependencies
+
+`ngbem` <br>
+`bemdom` <br>
+`$compile`
+
+#### Usage
+
+as element:
+
+```html
+<ng-bem
+    [bemjson="expression"]
+    [observe="string"]
+    [bemtree]>
+    [expression]
+</ng-bem>
+
+```
+
+as attribute:
+```html
+<ANY
+    ng-bem
+    [bemjson="expression"]
+    [observe="string"]
+    [bemtree]>
+    [expression]
+</ANY>
+
+```
+
+##### Arguments
+
+Param      | Type          | Details
+-----------|---------------|--------
+bemjson    | `expression`  | `BEMJSON` as [Angular Expression](https://docs.angularjs.org/guide/expression). Could be a refference to Object in scope.
+observe    | `string`      | Comma separated list of scope variable to observe. Changes on it would lead to rerendering. 
+bemtree    | `none`        | Flag to indicate whether `BEMTREE` templates should be used for `BEMJSON` processing.
+
+#### Example
+
+### bem
+
+Directives attaches to `i-bem.js` blocks using `data-bem` attribute and provides next set of features:
+ 
+ * Provide integration with some native Angular directives: `ngModel`, `ngChange` and events exposing directives like `ngBlur`
+ * Allow to bind `i-bem.js` block mod to settable angular expression through `bem-mod-*` attributes
+ * Call angular expression on `i-bem.js` events defined in `bem-event-*` and `bem-mod-event-*` attributes
+
+#### Dependencies
+
+`$parse`,
+`bemController`,
+`ngModelController`
+
+#### Usage
+
+You don't need to apply directive in your html templates.
+It just automatically applies to every `i-bem.js` block when angular applies $compile on some DOM nodes or HTML.
+Most likely you will use attributes provided by directive in `BEMJSON`.
+
+Note: currently we support only short syntax for attributes definitions.
+It means, that they applied to every block found on DOM-node.
+Support to explicit definition which block should be used planned on near releases.
```

Comment:
Support **for** explicit definition which block should be used **is** planned on near releases.